crypto3/libs/marshalling/zk/CMakeLists.txt

```diff
@@ -26,6 +26,7 @@ target_include_directories(${CMAKE_WORKSPACE_NAME}_${CURRENT_PROJECT_NAME} INTER
 
 target_link_libraries(${CMAKE_WORKSPACE_NAME}_${CURRENT_PROJECT_NAME} INTERFACE
                       Boost::container
+                      Boost::random
```

Comment:
may be make sense link `Boost` librarie only to executable (proof-producer and tests), not for header only libraries?
But we can do it in another PR

Reply:
i mean, no need to link boost libraries with header only targets

Reply:
But this way we keep track of which library uses which boost library. Doesn't help the compilation much, but probably does not hurt as well.